### PR TITLE
Update dev dependencies on `@fluidframework/eslint-config-fluid` in `azure` packages

### PR DESCRIPTION
**Pull request by Josmithr**
Also applies fixes.

### CODE REVIEW

**Comment by Josmithr** [comment]
@curtisman @tylerbutler layer-check is failing due to the dependency fix I made. Looks like the `external-controller` package might be mislabeled. Guidance for how to fix would be appreciated :)

File: azure/packages/external-controller/package.json [hunk 1]
```diff
@@ -44,14 +44,14 @@
   "dependencies": {
     "@fluidframework/azure-client": "^1.1.0",
     "@fluidframework/common-definitions": "^0.20.1",
+    "@fluidframework/test-client-utils": "^1.0.1",
```
Comment (tylerbutler) [inline azure/packages/external-controller/package.json:47]: This has to stay in the dev dependencies because of layer-check. See #10226.

Comment (Josmithr) [reply azure/packages/external-controller/package.json:47]: Okay. I'll revert the change then.

**Review by tylerbutler — APPROVED** [review]
The lockfile  shouldn't be added to the external-controller project since it's part of the release group.

**Comment by Josmithr** [comment]
> The lockfile shouldn't be added to the external-controller project since it's part of the release group.

Ah, for some reason I thought we still had them per package. My mistake. I have removed it.

**Pull request merged**